redesign/base_generator.py

```diff
@@ -1,21 +1,18 @@
 """Contains abstract class from which generators will inherit."""
 
 import abc
+from typing import Dict
 
 
 class BaseGenerator(metaclass=abc.ABCMeta):
     """Abstract generator class."""
 
     def __init__(self, profile, options):
         """Initialize generator object."""
-        self.profile = profile.report()
-
-        if isinstance(options, dict):
-            self.options = options
-        else:
-            self.options = {}
+        self.report = profile.report()
+        self.options = options
 
     @abc.abstractmethod
-    def run(self):
+    def synthesize(self, options: Dict=None):
```

Comment:
the issue with using options here is that a user wouldn't know exactly what options should be. Maybe we shouldn't use options at all and should just use the actual names so it is declarative and obvious to the user.

Reply:
Thoughts?

Reply:
This abstract API should match below, yes?

Reply:
I think that would be better since there wouldn't be that many arguments.

Reply:
I thought of an issue, being those declarative values maybe diff per generator. but doesn't this run into the same issues as the DP and the data processing abstraction?

Reply:
It was LSP
https://stackoverflow.com/questions/14626279/inheritance-best-practice-args-kwargs-or-explicitly-specifying-parameters